app/services/migration/migrator.rb

```diff
@@ -39,25 +41,69 @@ def check_environment!
 
     def run_migration
       migrate_lead_providers
+      migrate_cohorts
+      migrate_statements
     end
 
-    def migrate_lead_providers
-      data_migration = DataMigration.find_by(model: :lead_provider)
-
-      data_migration.update!(started_at: Time.zone.now, total_count: ecf_npq_lead_providers.count)
+    def migrate(items, model)
+      data_migration = DataMigration.find_by(model:)
+      data_migration.update!(started_at: Time.zone.now, total_count: items.count)
 
-      ecf_npq_lead_providers.find_each do |ecf_npq_lead_provider|
+      items.each do |item|
         data_migration.increment!(:processed_count)
-        npq_lead_provider = LeadProvider.find_by(ecf_id: ecf_npq_lead_provider.id)
-
-        data_migration.increment!(:failure_count) unless npq_lead_provider
+        result = yield(item)
+        data_migration.increment!(:failure_count) unless result
       end
 
       data_migration.update!(completed_at: Time.zone.now)
     end
 
+    def migrate_lead_providers
+      migrate(ecf_npq_lead_providers, :lead_provider) do |ecf_npq_lead_provider|
+        LeadProvider.find_by(ecf_id: ecf_npq_lead_provider.id)
+      end
+    end
+
     def ecf_npq_lead_providers
       @ecf_npq_lead_providers ||= Ecf::NpqLeadProvider.all
     end
+
+    def ecf_cohorts
+      @ecf_cohorts ||= Ecf::Cohort.where(start_year: 2021..)
+    end
+
+    def migrate_cohorts
+      migrate(ecf_cohorts, :cohort) do |ecf_cohort|
+        cohort = Cohort.find_or_initialize_by(start_year: ecf_cohort.start_year)
+        result = cohort.update(registration_start_date: ecf_cohort.registration_start_date)
```

Comment:
```suggestion
        result = cohort.update(registration_start_date: ecf_cohort.npq_registration_start_date)
```

Reply:
some of those are `nil`, do we want to check the logic in ECF around how those defaulted before 2023 and maybe then populating with `registration_start_date`?

Reply:
HI @cwrw I can only see `npq_registration_start_date` being used to fetch the current `active_npq_registration_cohort` when building a npq application in ECF app.. not being used in the "past" so maybe that's why we have nil `npq_registration_start_date` value for past cohorts..